src/validator/has-length-equal-to.ts

```diff
@@ -0,0 +1,37 @@
+import {STRule} from '../rule/rule';
+import {STRuleFn} from '../rule/fn';
+import {STRuleModifiers} from '../rule/modifiers';
+import {STRuleNode} from '../rule/node';
+import {STRuleNodeType} from '../rule/node-type';
+import {equalToFn} from './is-equal-to';
+
+export type STOpHasLengthEqualTo<CallerType> = (a: number) => CallerType;
+
+export const hasLengthEqualTo = (curr: any, target: number): boolean => {
+	if (curr.length <= -1) {
```

Comment:
This can be simplified a little to be:
```typescript
if (curr.length < 0) {
```
After changing the !curr.length check below, move this check below it. We always perform type tests prior to comparisons or value operations. The reason is simple: if a safety check is worth performing on a type to make sure it matches the expected type, then something bad can happen if the type is wrong. If the type is wrong, using comparisons, comparators, or any other operators can produce indeterminate behavior, so we haven no idea what it will do.